utilities/src/main/java/org/craftercms/commons/file/blob/impl/BlobStoreResolverImpl.java

```
@@ -0,0 +1,115 @@
+        this.configurationPath = configurationPath;
+    }
+
+    public void setConfigurationReader(EncryptionAwareConfigurationReader configurationReader) {
+        this.configurationReader = configurationReader;
+    }
+
+    protected BlobStore findStore(HierarchicalConfiguration config, Predicate<HierarchicalConfiguration> predicate)
+            throws ConfigurationException {
+        Optional<HierarchicalConfiguration> storeConfig =
+                config.configurationsAt(CONFIG_KEY_STORE).stream().filter(predicate).findFirst();
+        if (storeConfig.isPresent()) {
+            HierarchicalConfiguration store = storeConfig.get();
+            String type = store.getString(CONFIG_KEY_TYPE);
+            BlobStore instance = applicationContext.getBean(type, BlobStore.class);
+            instance.init(store);
+            return instance;
+        }
+        return null;
+    }
+
+    protected HierarchicalConfiguration getConfiguration(Function<String, InputStream> configGetter) {
+        logger.debug("Reading blob store configuration");
+        try (InputStream is = configGetter.apply(configurationPath)) {
+            if (is != null) {
+                return configurationReader.readXmlConfiguration(is, "utf-8");
+            }
+        } catch (Exception e) {
+            logger.error("Error reading blob store configuration", e);
+        }
+        return null;
+    }
+
+    @Override
+    public BlobStore getById(Function<String, InputStream> configGetter, String storeId) {
+        logger.debug("Looking blob store with id {}", storeId);
+        try {
+            HierarchicalConfiguration config = getConfiguration(configGetter);
+            if (config != null) {
+                return findStore(config, store ->
+                        StringUtils.equals(storeId, store.getString(CONFIG_KEY_ID)));
+            }
+        } catch (ConfigurationException e) {
+            logger.error("Error reading blob store configuration", e);
```

Why are we swallowing the exception? Isn't this something the caller should know?